pkg/apis/telemetry/v1alpha1/types.go

```diff
@@ -0,0 +1,201 @@
+	// Default value depends on the kind of the component.
+	// +optional
+	Name string `json:"name,omitempty"`
+	// Version defines the version of the chart.
+	// Default value depends on the kind of the component.
+	// +optional
+	Version string `json:"version,omitempty"`
+}
+
+type ThanosConfig struct {
+	ComponentConfig `json:",inline"`
+	// Mode defines the mode of the Thanos.
+	// The default is sidecar.
+	// +kubebuilder:validation:Enum=sidecar;reciever
+	// +optional
+	Mode string `json:"mode,omitempty"`
+	// ObjectStoreConfig defines the configuration for the object store.
+	ObjectStore *ThanosObjectStoreConfig `json:"objectStore,omitempty"`
+	// ExtraArgs is the set of extra arguments for Thanos chart.
+	//
+	// For Example, using following configuration to enable query frontend.
+	// extraArgs:
+	//   queryFrontend:
+	//	   enabled: true
+	//
+	// +optional
+	ExtraArgs apiextensionsv1.JSON `json:"extraArgs,omitempty"`
+}
+
+type ThanosObjectStoreConfig struct {
+	// Type defines the type of the object store.
+	// Now only support s3, which means S3-compatible object storages, e.g. MinIO.
+	Type string `json:"type"`
+	// S3 defines the configuration for the S3 object store.
+	// For more information, see https://thanos.io/tip/thanos/storage.md/#s3
+	// This field is required if the type is s3.
+	// +optional
+	S3 *S3Config `json:"s3,omitempty"`
+}
+
+// FilesystemConfig defines the configuration for storing and accessing blobs in filesystem.
+type FilesystemConfig struct {
+	Directory string `json:"directory"`
+}
+
+type S3Config struct {
+	// Bucket defines the name of the bucket.
+	Bucket string `json:"bucket"`
+	// Region defines the region of the bucket.
+	Region string `json:"region"`
+	// Endpoint defines the endpoint of the bucket.
+	Endpoint string `json:"endpoint"`
+	// Credential is the credential used to access the bukcet.
+	// Make sure the credential is in the same namespace as the monitoring.
+	// The secret must have following keys: AccessKeyID, SecretAccessKey.
+	// SessionToken is optional.
+	Credential clusterv1alpha1.CredentialConfig `json:"credential"`
+}
+
+type GrafanaConfig struct {
+	ComponentConfig `json:",inline"`
+	// ExtraArgs is the set of extra arguments for Grafana chart.
+	//
+	// For Example, using following configuration to change replica count.
+	// extraArgs:
+	//   grafana:
+	//     replicaCount: 2
+	//
+	// +optional
+	ExtraArgs apiextensionsv1.JSON `json:"extraArgs,omitempty"`
+}
+
+// TelemetryStatus defines the observed state of Telemetry.
+type TelemetryStatus struct {
+	// Phase represents the current phase of Telemetry.
+	// E.g. Pending, Running, Terminating, Failed, Ready, etc.
+	// +optional
+	Phase string `json:"phase,omitempty"`
+	// A brief message indicating details about why the Telemetry is in this state.
+	// +optional
+	Reason string `json:"reason,omitempty"`
+	// Endpoints are the endpoints for telemetry components.
+	Endpoints Endpoints `json:"querierEndpoints,omitempty"`
```

Comment:
Endpoints do not sound very right here. It is actually metric scrape endpoint right?

And since we have a aggregation server, so this is just a hint to user, which are not needed by user